YQL-17087: Enable output channel spilling only if task has two or more outputs

Darych opened the pull request.

Darych committed:
Enable output channel spilling only if task has two or more outputs

Darych committed:
Enable output channel spilling only if task has two or more outputs

rvu1024 commented on ydb/library/yql/providers/dq/api/protos/dqs.proto:
```diff
@@ -39,7 +39,6 @@ message TAllocateWorkersRequest {
 
     uint64 FreeWorkerAfterMs = 14;
     NYql.NDqProto.EDqStatsMode StatsMode = 16;
-    bool EnableSpilling = 17; // false
```
Нельзя простj так удалять - это протобуф используется для вызовов api. Нужно переносить в reserved

rvu1024 reviewed (changes requested):
Как будет реагировать сейчас код на ситуацию, когда spilling service выключен, а в запросе ставят прагму dq.SpillingEngine="file"?

Darych committed:
Address comments

Darych committed:
Address comments

Darych commented:
> Как будет реагировать сейчас код на ситуацию, когда spilling service выключен, а в запросе ставят прагму dq.SpillingEngine="file"?

Если в запросе реально начнет применяться спиллинг, то запрос зависнет, потому что данные никогда не запишутся, а следовательно не прочитаются.

Darych committed:
Add handling if spilling service is not started.
TDqLocalFileSpillingActor sends events with flag to traack undelivery.
Add unit tests for this.
Fix LOG macroses in channel_storage_actor.cpp to handle ActorSystem pointer also.

Darych committed:
Add handling if spilling service is not started.
TDqLocalFileSpillingActor sends events with flag to traack undelivery.
Add unit tests for this.
Fix LOG macroses in channel_storage_actor.cpp to handle ActorSystem pointer also.

Darych commented:
> Как будет реагировать сейчас код на ситуацию, когда spilling service выключен, а в запросе ставят прагму dq.SpillingEngine="file"?

Добавил флажок для отслеживания Undelivered. Spilling actor будет следить за undelivered и в случае недоставки ответит EvError.

lll-phill-lll commented:
А теперь получается, что включение спиллинга не всегда включает спиллинг?

Darych committed:
Return backward compatibility to YDB for spilling

Darych committed:
Return backward compatibility to YDB for spilling

Darych commented:
> А теперь получается, что включение спиллинга не всегда включает спиллинг?

Для YDB пока еще всегда. Для обратной совместимости с прошлой версией.
В  DQ уже спиллинг в каналах будет происходить только когда из стейджа выходит больше одного канала.

Darych committed:
Fix backward compatibility for YDB
